include/swift/Basic/LangOptions.h

```diff
@@ -288,6 +288,10 @@ namespace swift {
     /// Whether to verify the parsed syntax tree and emit related diagnostics.
     bool VerifySyntaxTree = false;
 
+    /// Scaffolding to permit experimentation with finer-grained dependencies
+    /// and faster rebuilds.
+    bool EnableExperimentalDependencies = false;
```

Comment:
Another option is to put this on a SourceFile's ReferencedNameTracker. I'm not sure if that's better since it means putting this flag in FrontendOptions just to pass it through, but it might be easier for whatever implementation you end up with to have the flag right there. Since that's inside the frontend, though, you can always change it later.

Reply:
Good thought, will keep it in mind.